agent/agentssh/agentssh.go

```diff
@@ -82,7 +99,7 @@ type Server struct {
 	metrics *sshServerMetrics
 }
 
-func NewServer(ctx context.Context, logger slog.Logger, prometheusRegistry *prometheus.Registry, fs afero.Fs, maxTimeout time.Duration, x11SocketDir string) (*Server, error) {
+func NewServer(ctx context.Context, logger slog.Logger, prometheusRegistry *prometheus.Registry, fs afero.Fs, config *Config) (*Server, error) {
```

Comment:
If we're extracting some of the args to `NewServer` to a struct, why not go all the way and add the logger, registry, and fs too?

Reply:
We could combine them, but I wanted to keep these separate to better mimic a `sshd_config`. The config manages optional knobs, but logger/prom/fs are more deeply integrated in the server and not optional.

Reply:
That's fair, and placing them in the config does imply optionality.